src/app/audiologist-summary/audiologist-summary.component.ts

```
@@ -0,0 +1,231 @@
+  }
+
+  public setTS() {
+    let answers = this.tsDataService.dataRecord;
+    let tsAnswers = new TsScreenerAnswerStrings();
+    let index = 0;
+    if (answers.length < 1) {
+      return;
+    }
+    if (answers[index++].choice === tsAnswers.NO) {
+      this.ts = "No Tinnitus";
+      return;
+    }
+
+    this.ts = (answers[index++].choice === tsAnswers.NO) ? "Acute" : "Chronic";
+    if (answers[index++].choice !== tsAnswers.SOMETIMES_OCCASIONALLY) {
+      this.ts += " Constant Tinnitus";
+      return;
+    }
+    if (answers[index++].choice === tsAnswers.YES_ALWAYS) {
+      this.ts += " Temporary Tinnitus";
+      return;
+    }
+    if (answers[3].choice === tsAnswers.YES_SOMETIMES) {
+      if (answers[index++].choice === tsAnswers.NO) {
+        this.ts += " Temporary Tinnitus";
+        return;
+      }
+    }
+    this.ts += (answers[index].choice === tsAnswers.DAILY_OR_WEEKLY_BASIS) ? " Intermittent Tinnitus" : " Occasional Tinnitus";
+  }
+
+  public setTHS() {
+    let answers = this.thsDataService.dataRecord;
+    if (answers.length < 1) {
+      return;
+    }
+    this.thsA = this.sumTHS(answers, 0, 4);
+    this.thsB = this.sumTHS(answers, 4, 4);
+    this.thsC = this.getTHSvalue(answers, 8);
+    this.thsCtxt = answers[8].choice;
+    this.thsCex = this.thsC > 0 ? answers[9].choice : "";
+    console.log(answers);
+  }
+  private sumTHS(array, start, length): number {
+    let sum: number = 0;
+    length += start;
+    for (let i = start; i < length; i++) {
+      sum += this.getTHSvalue(array, i);
+    }
+    return sum;
+  }
+  private getTHSvalue(array, i): number {
+    let thsAnswers = new ThsAnswerStrings();
+    if (array.length < 1) {
+      return;
+    }
+    switch (array[i].choice) {
+      case thsAnswers.NO: return 0;
+      case thsAnswers.SMALL_YES: return 1;
+      case thsAnswers.MODERATE_YES: return 2;
+      case thsAnswers.BIG_YES: return 3;
+      case thsAnswers.VERY_BIG_YES: return 4;
+      default: return 0;
```

This should probably throw an error instead of returning nonsense.

---

The above makes this N/A